impeller/entity/contents/linear_gradient_contents.cc

```diff
@@ -63,6 +59,10 @@ bool LinearGradientContents::Render(const ContentContext& renderer,
     }
   }
 
+  auto placeholder = Entity();
+  auto gradient_snapshot =
+      gradient_generator_->RenderToSnapshot(renderer, placeholder);
```

Comment:
Should we try to elide this if there are only two colors? This will mean we're always doing a RT switch even when we wouldn't otherwise need to, right?

Reply:
I added special two color versions. From some cursory searching in google3, almost all linear gradients are two color